src/compile/common.ts

```diff
@@ -30,8 +30,7 @@ export function buildModel(spec: Spec, parent: Model, parentGivenName: string):
 }
 
 // TODO: figure if we really need opacity in both
-export const STROKE_CONFIG = ['stroke', 'strokeWidth',
-  'strokeDash', 'strokeDashOffset', 'strokeOpacity', 'opacity'];
+export const STROKE_CONFIG = ['stroke', 'strokeWidth', 'strokeOpacity', 'opacity'];
```

Comment:
I don't think this is the right fix.  This would make strokeDash unused for both legend and marks.  

Why don't you modify the Line 128 in legend.ts like I suggested?